ClientApp/KachnaOnline/src/app/events/event-form/event-form.component.ts

```diff
@@ -148,7 +153,24 @@ export class EventFormComponent implements OnInit {
     }
   }
 
-  planEvent(eventData: Event) {
+  private verifyDates(eventData: EventModification) {
+    if ((new Date(eventData.from)).getTime() < Date.now()) {
+      this.toastr.error("Akce nemůže začínat v minulosti. Upravte termín počátku akce.", "Plánování akce")
+      return false;
+    }
+    if ((new Date(eventData.from)).getTime() < Date.now()) {
+      this.toastr.error("Akce nemůže končit v minulosti. Upravte termín konce akce.", "Plánování akce")
+      return false;
+    }
+    if ((new Date(eventData.from)) >= (new Date(eventData.to))) {
+      this.toastr.error("Akce nemůže začínat po jejím konci. Upravte termín počátku nebo konce akce.", "Plánování akce")
+      return false;
+    }
```

Comment:
Once again, you're making `Date` objects based on a 'time zone adjusted' string that you've assembled from the form model before – and comparing it to the current UTC timestamp. This is bound to introduce weird bugs. I would add a method like this:

```ts
private joinDateTime(date: NgbDateStruct, time: NgbTimeStruct) : Date | null {
  let dateObj = this.nativeDateAdapter.toModel(date);
  dateObj?.setHours(time.hour);
  dateObj?.setMinutes(time.minute);
  dateObj?.setSeconds(0);
  return dateObj;
}
```

Then you can compare the mapped Date with the current timestamp:

```ts
const from = this.joinDateTime(formVal.fromDate, formVal.fromTime);

if (from.getTime() < Date.now()) { ... }
```

... and it should, in theory, be _safe_:tm:.

Reply:
I will do as you say. It makes sense to me, so let us hope it will truly be working as intended.